programming_challenges/hacker_rank/eval_ex/main.clj

```diff
@@ -0,0 +1,70 @@
+;; https://www.hackerrank.com/challenges/eval-ex/problem
+
+;; Approach:
+;; Use map to transform the list of 10 elements from x to (/ (exp x n) (factorial n))
+;; And reduce suming each element
+
+;; For this approach, we can separate in 3 different functions
+;; - exp: exponantial
+;; - factorial
+;; - expansion: function composed of exp and factorial passed to map
+
+;; So we can test these 3 functions
+
+(ns eval-ex
+  (:use [clojure.test]))
+
+(def ten-terms (range 10))
+
+(defn exp
+  [x n]
+  (reduce * (repeat n x)))
+
+(defn factorial
+  [n]
+  (reduce * (range 1 (inc n))))
+
+(defn expansion
+  [x n]
+  (/ (exp x n) (factorial n)))
+
+(defn series-expansion
+  [x]
+  (read-string
+    (format "%.4f"
+      (reduce
+        +
+        (map
+          (fn [n] (expansion x n))
```

Comment:
It's probably a bit more idiomatic to say `(partial expansion x)` here but another possibility is to define `expansion` as a "curried" function in the first place:
```
(defn expansion
  [x]
  (fn [n]
    (/ (exp x n) (factorial n))))
```
then you can call `(expansion x)` to get back a function (of `n`) so you can `(map (expansion x) ten-terms)`

Reply:
That means your tests would need to change to, e.g., `(is (= 1 ((expansion 10) 0)))`